status/condition_set.go

```diff
@@ -237,3 +239,62 @@ func (c ConditionSet) findUnhealthyDependents() []Condition {
 	})
 	return conditions
 }
+
+// unstructuredAdapter is an adapter for the status.Object interface. unstructuredAdapter
+// makes the assumption that status conditions are found on status.conditions path.
+type unstructuredAdapter struct {
+	*unstructured.Unstructured
+}
+
+// FromUnstructured makes the assumption that the status conditions are found on status.conditions
+// path. If they are not found then we return nil.
+func FromUnstructured(u *unstructured.Unstructured) (Object, error) {
+	c, found, err := unstructured.NestedSlice(u.Object, "status", "conditions")
+	if err != nil || !found {
+		return nil, fmt.Errorf("unable to determine status conditions: %w", err)
```

Comment:
Thoughts on panicking, since we control the types that we call this with? This will simplify callers significantly.

Reply:
I'm wondering if since this is a library if we should just be comfortable with throwing an error here and then letting the caller control whether they want to panic or not. I'm realistically fine with either; just thinking that errors give you a bit more flexibility

Reply:
> Thoughts on panicking, since we control the types that we call this with? This will simplify callers significantly.

I have seen that the caller gets a bit complicated if we don't panic since there's some error handling that has to be done. 
But I also agree with @jonathan-innis that caller can decide what needs to be done in case of an error.